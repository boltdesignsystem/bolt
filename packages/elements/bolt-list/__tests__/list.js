// Refer to https://github.com/boltdesignsystem/bolt/wiki/Jest-Test:-Example-Jest-Test for more testing examples
import { render, stopServer } from '../../../testing/testing-helpers';
import schema from '../list.schema';
const { tag, display, spacing, align, valign, separator } = schema.properties;
let page, fixtures;

afterAll(async () => {
  await stopServer();
  await page.close();
});

beforeEach(async () => {
  await page.evaluate(() => {
    document.body.innerHTML = '';
  });
});

beforeAll(async () => {
  page = await global.__BROWSER__.newPage();
  await page.goto('http://127.0.0.1:4444/', {
    timeout: 0,
  });

  const defaultData = {
    content: 'This is a list',
  };

  fixtures = {
    defaultData,
  };
});

describe('Bolt List', () => {
  test('default', async () => {
    const results = await render('@bolt-elements-list/list.twig', {
      ...fixtures.defaultData,
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});

describe('Bolt List prop -', () => {
  // Target each of the schema keys with the following pattern
  tag.enum.forEach(async option => {
    test(`tag: ${option}`, async () => {
      const results = await render('@bolt-components-list/list.twig', {
        ...fixtures.defaultData,
        tag: option,
        items: ['item 1', 'item 2', 'item 3'],
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
  display.enum.forEach(async option => {
    test(`display: ${option}`, async () => {
      const results = await render('@bolt-components-list/list.twig', {
        ...fixtures.defaultData,
        display: option,
        items: ['item 1', 'item 2', 'item 3'],
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
  spacing.enum.forEach(async option => {
    test(`spacing: ${option}`, async () => {
      const results = await render('@bolt-components-list/list.twig', {
        ...fixtures.defaultData,
        spacing: option,
        items: ['item 1', 'item 2', 'item 3'],
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
  align.enum.forEach(async option => {
    test(`horizontal align: ${option}`, async () => {
      const results = await render('@bolt-components-list/list.twig', {
        ...fixtures.defaultData,
        align: option,
        items: ['item 1', 'item 2', 'item 3'],
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
  valign.enum.forEach(async option => {
    test(`vertical align: ${option}`, async () => {
      const results = await render('@bolt-components-list/list.twig', {
        ...fixtures.defaultData,
        valign: option,
        items: ['item 1', 'item 2', 'item 3'],
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
  separator.enum.forEach(async option => {
    test(`separator: ${option}`, async () => {
      const results = await render('@bolt-components-list/list.twig', {
        ...fixtures.defaultData,
        separator: option,
        items: ['item 1', 'item 2', 'item 3'],
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
  [true, false].forEach(async insetOption => {
    test(`inset: ${insetOption}`, async () => {
      const results = await render('@bolt-components-list/list.twig', {
        ...fixtures.defaultData,
        inset: insetOption,
        items: ['item 1', 'item 2', 'item 3'],
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
  [true, false].forEach(async nowrapOption => {
    test(`no wrapping: ${nowrapOption}`, async () => {
      const results = await render('@bolt-components-list/list.twig', {
        ...fixtures.defaultData,
        nowrap: nowrapOption,
        items: ['item 1', 'item 2', 'item 3'],
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
  [true, false].forEach(async fullWidthOption => {
    test(`full width option: ${fullWidthOption}`, async () => {
      const results = await render('@bolt-components-list/list.twig', {
        ...fixtures.defaultData,
        inline_full_width: fullWidthOption,
        items: ['item 1', 'item 2', 'item 3'],
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
});
