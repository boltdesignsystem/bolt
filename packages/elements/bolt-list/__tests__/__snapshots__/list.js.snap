// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt List default 1`] = `
<ul class="e-bolt-list">
  This is a list
</ul>
`;

exports[`Bolt List prop - display: horizontal 1`] = `
<ul class="e-bolt-list e-bolt-list--display-horizontal">
  This is a list
</ul>
`;

exports[`Bolt List prop - display: horizontal@medium 1`] = `
<ul class="e-bolt-list e-bolt-list--display-horizontal@medium">
  This is a list
</ul>
`;

exports[`Bolt List prop - display: horizontal@small 1`] = `
<ul class="e-bolt-list e-bolt-list--display-horizontal@small">
  This is a list
</ul>
`;

exports[`Bolt List prop - display: horizontal@xsmall 1`] = `
<ul class="e-bolt-list e-bolt-list--display-horizontal@xsmall">
  This is a list
</ul>
`;

exports[`Bolt List prop - display: horizontal@xxsmall 1`] = `
<ul class="e-bolt-list e-bolt-list--display-horizontal@xxsmall">
  This is a list
</ul>
`;

exports[`Bolt List prop - display: vertical 1`] = `
<ul class="e-bolt-list">
  This is a list
</ul>
`;

exports[`Bolt List prop - horizontal align: center 1`] = `
<ul class="e-bolt-list e-bolt-list--align-center">
  This is a list
</ul>
`;

exports[`Bolt List prop - horizontal align: end 1`] = `
<ul class="e-bolt-list e-bolt-list--align-end">
  This is a list
</ul>
`;

exports[`Bolt List prop - horizontal align: justify 1`] = `
<ul class="e-bolt-list e-bolt-list--align-justify">
  This is a list
</ul>
`;

exports[`Bolt List prop - horizontal align: start 1`] = `
<ul class="e-bolt-list">
  This is a list
</ul>
`;

exports[`Bolt List prop - inset: false 1`] = `
<ul class="e-bolt-list">
  This is a list
</ul>
`;

exports[`Bolt List prop - inset: true 1`] = `
<ul class="e-bolt-list e-bolt-list--inset">
  This is a list
</ul>
`;

exports[`Bolt List prop - no wrapping: false 1`] = `
<ul class="e-bolt-list">
  This is a list
</ul>
`;

exports[`Bolt List prop - no wrapping: true 1`] = `
<ul class="e-bolt-list e-bolt-list--nowrap">
  This is a list
</ul>
`;

exports[`Bolt List prop - separator: dashed 1`] = `
<ul class="e-bolt-list e-bolt-list--separator-dashed">
  This is a list
</ul>
`;

exports[`Bolt List prop - separator: none 1`] = `
<ul class="e-bolt-list">
  This is a list
</ul>
`;

exports[`Bolt List prop - separator: solid 1`] = `
<ul class="e-bolt-list e-bolt-list--separator-solid">
  This is a list
</ul>
`;

exports[`Bolt List prop - spacing: large 1`] = `
<ul class="e-bolt-list e-bolt-list--spacing-large">
  This is a list
</ul>
`;

exports[`Bolt List prop - spacing: medium 1`] = `
<ul class="e-bolt-list e-bolt-list--spacing-medium">
  This is a list
</ul>
`;

exports[`Bolt List prop - spacing: none 1`] = `
<ul class="e-bolt-list e-bolt-list--spacing-none">
  This is a list
</ul>
`;

exports[`Bolt List prop - spacing: small 1`] = `
<ul class="e-bolt-list e-bolt-list--spacing-small">
  This is a list
</ul>
`;

exports[`Bolt List prop - spacing: xlarge 1`] = `
<ul class="e-bolt-list e-bolt-list--spacing-xlarge">
  This is a list
</ul>
`;

exports[`Bolt List prop - spacing: xsmall 1`] = `
<ul class="e-bolt-list">
  This is a list
</ul>
`;

exports[`Bolt List prop - spacing: xxsmall 1`] = `
<ul class="e-bolt-list e-bolt-list--spacing-xxsmall">
  This is a list
</ul>
`;

exports[`Bolt List prop - tag: ol 1`] = `
<ol class="e-bolt-list">
  This is a list
</ol>
`;

exports[`Bolt List prop - tag: ul 1`] = `
<ul class="e-bolt-list">
  This is a list
</ul>
`;

exports[`Bolt List prop - vertical align: center 1`] = `
<ul class="e-bolt-list">
  This is a list
</ul>
`;

exports[`Bolt List prop - vertical align: end 1`] = `
<ul class="e-bolt-list e-bolt-list--valign-end">
  This is a list
</ul>
`;

exports[`Bolt List prop - vertical align: start 1`] = `
<ul class="e-bolt-list e-bolt-list--valign-start">
  This is a list
</ul>
`;
